src/lib/device_tree/src/lib.rs

```diff
@@ -203,7 +203,10 @@ pub fn infer_type(data: &[u8]) -> Type {
     }
     if let Some(i) = data.iter().position(|&c| !is_print(c)) {
         if i == data.len() - 1 && data[i] == 0 {
```

Comment:
we were wondering. Can we drop the iterator? Since there's one in the from_utf8 anyway.

if data[data.len()-1] != 0 {
return something;
}

            match core::str::from_utf8(&data[..data.len() - 1]) {
                Ok(ret) => return Type::String(ret),
                Err(_e) => {  },
            }

Reply:
Yep, updated.

Reply:
Nope! It looks like it is required. This does some strange value checking to ensure we don't over convert numbers to strings.

Without this I see prints like this:
>         area@8
>           description = "Empty Space"
>           offset = "\u{1}\u{0}\u{0}"
>           size = "\u{1}\u{0}\u{0}"